services/brig/src/Brig/User/Search/Index.hs

```diff
@@ -827,9 +862,17 @@ reindexRowToIndexUser
 
       idpUrl :: UserSSOId -> Maybe Text
       idpUrl (UserSSOId (SAML.UserRef (SAML.Issuer uri) _subject)) =
-        Just $ (cs . toLazyByteString . serializeURIRef) uri
+        Just $ fromUri uri
       idpUrl (UserScimExternalId _) = Nothing
 
+      fromUri :: URI -> Text
+      fromUri = cs . toLazyByteString . serializeURIRef
+
+      sso :: UserSSOId -> Maybe Sso
+      sso userSsoId = do
+        (issuer, nameid) <- User.ssoIssuerAndNameId userSsoId
```

Comment:
[optional] Why the detour through the tuple?  I would have named `Sso` `SsoOnly` (or something else more specific) and typed `User.ssoIssuerAndNameId :: UserSSOId -> Maybe SsoOnly`.

Reply:
Yes, but this would make a cyclic dependency, therefore the tuple.